src/GSF.CarbonAware/test/ForecastHandlerTests.cs

```diff
@@ -38,34 +37,65 @@ public async Task GetCurrentAsync_Succeed_Call_MockAggregator_WithOutputData(str
             }
         };
 
-        var aggregator = SetupMockAggregator(data);
+        var aggregator = CreateCurrentForecastAggregator(data);
         var handler = new ForecastHandler(Logger!.Object, aggregator.Object);
         var locations = location2 != null ? new string[] { location1, location2 } : new string[] { location1 };
         var result = await handler.GetCurrentForecastAsync(locations, start, end, duration);
         Assert.That(result, Is.Not.Empty);
     }
 
     [Test]
-    public async Task GetCurrentAsync_Succeed_Call_MockAggregator_WithoutOutputData()
+    public async Task GetCurrentForecastAsync_Succeed_Call_MockAggregator_WithoutOutputData()
     {
-        var aggregator = SetupMockAggregator(Array.Empty<global::CarbonAware.Model.EmissionsForecast>().ToList());
+        var aggregator = CreateCurrentForecastAggregator(Array.Empty<global::CarbonAware.Model.EmissionsForecast>().ToList());
         var handler = new ForecastHandler(Logger!.Object, aggregator.Object);
         var result = await handler.GetCurrentForecastAsync(new string[] { "eastus" }, DateTimeOffset.Now, DateTimeOffset.Now + TimeSpan.FromHours(1), 30);
         Assert.That(result, Is.Empty);
     }
 
+    [TestCase("Sydney","2022-03-07T01:00:00", "2022-03-07T03:30:00", "2022-03-07T03:30:00", 5, TestName = "GetForecastByDate calls aggregator: all fields")]
+    [TestCase("Sydney", null, null, "2022-03-07T01:00:00", null, TestName = "GetForecastByDate calls aggregator: required fields only")]
+    public async Task GetForecastByDateAsync_Succeed_Call_MockAggregator_WithOutputData(string location, DateTimeOffset? start, DateTimeOffset? end, DateTimeOffset requestedAt, int? duration)
+    {
+        var data = new global::CarbonAware.Model.EmissionsForecast {
```

Comment:
qq: Do you need to specify the namespace this way? Is there any namespace conflict somewhere?

Reply:
Yeah we do. Because the model name in CarbonAware.Models matches the name in GSF.CarbonAware.Models so we need to distinguish when we need the models from the aggregator return type vs. the library return type